inspirehep/modules/orcid/utils.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""ORCID utils."""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import re
+import requests
+
+from orcid import MemberAPI
+
+from inspire_utils.config import load_config
+
+from .converter import OrcidConverter
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def push_record_with_orcid(recid, orcid, oauth_token, put_code=None):
+    """Push record to ORCID with a specific ORCID ID.
+
+    Args:
+        recid (string): HEP record to push
+        orcid (string): ORCID identifier to push onto
+        oauth_token (string): ORCID user OAUTH token
+        put_code (Union[string, NoneType]): put-code to push record onto,
+            if None will push as a new record
+
+    Returns:
+        string: the put-code of the updated/inserted item
+    """
+    config = load_config()
+
+    server_name = config['SERVER_NAME']
+
+    if not re.match('^https?://', server_name):
```

Comment:
Maybe this should be tested as well with a case of a server name url not containing http.